arangod/Agency/Agent.cpp

```diff
@@ -1960,10 +1961,16 @@ void Agent::emptyCbTrashBin() {
     _callbackLastPurged = std::chrono::steady_clock::now();
   }
 
-  LOG_TOPIC(DEBUG, Logger::AGENCY) << "unobserving: " << envelope->toJson();
-
-  // Best effort. Will be retried anyway
-  auto wres = write(envelope);
+  // Best effort. Will be retried otherwise.
+  LOG_TOPIC(DEBUG, Logger::AGENCY) << "scheduling unobserve: " << envelope->toJson();
+  auto* scheduler = SchedulerFeature::SCHEDULER;
+  std::shared_ptr<Agent> agent = this->shared_from_this();
+  if (scheduler != nullptr) {
+    scheduler->queue(
```

Comment:
Queuing may fail (in this case `false` is returned by `Scheduler::queue()`).
This case is not handled here, but it should be.